### ISSUE TITLE
Updater doesn't recognize .10 as being higher than .9

### ISSUE THREAD

**Issue by GoogleCodeExporter** [hoanglanton/cyanogen-updater#155]
```
First this is not a Bugtracker for Cyanogen-Rom Bugs. Only for the updater.
The other Bugtracker can be found here:
http://code.google.com/p/cyanogenmod/issues/list

What steps will reproduce the problem?
1. Open cm-updater when running any releases with minor numbers ending in 
.1-9 (i.e. 4.2.8)
2. Check for update

What is the expected output? What do you see instead?
Expect to see 4.2.10.1, the latest rom

What ROM are you running?(including Version)
4.2.9.1

What Version of CM-Updater are you running??(When using an Experimental
one, please Provide the SVN Release Number)
4.5.1


Please provide any additional information below.
As mentioned on twitter by Steve, enabling "show all releases" will reveal 
the 4.2.10.1 release.  It's probably only looking at one digit instead of 
two, so .10 looks like .1 to it.
```

Original issue reported on code.google.com by `patrickj...@gmail.com` on 24 Dec 2009 at 8:25

**Comment by GoogleCodeExporter** [comment]
```
I don't think it's only looking at one digit, as it does properly increment 
.10, .11, 
.12, etc.  It looks to be numbering things much like windows used to (e.g., 
img1, 
img10, img11, img2, img20, img21, img3, img4, img5).  It's essentially treating 
each 
digit as a sub-version, rather than waiting for a period to define sub-versions.
```

Original comment by `neatc...@gmail.com` on 25 Dec 2009 at 5:44

**Comment by GoogleCodeExporter** [comment]
```
thats because the version string is String Compared (Java Method). I have to do 
this, cause there are themes out 
there with versions like 4.3.3r3 and so on. If you know a better way to check 
versions, please mail me
```

Original comment by `FireFart` on 25 Dec 2009 at 6:00

**Comment by GoogleCodeExporter** [comment]
```
Someone in #cyanogenmod suggested using an auto-incrementing version number in 
the json 
that is not exposed to the user.  Would obviously have to collaborate with 
Steve on 
that, but a possibility if no other solution can be found?
```

Original comment by `neatc...@gmail.com` on 25 Dec 2009 at 6:32

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
@firefart
One way I can think of is strip all the alphabets n dots. Then convert the 
resultant value to numbers and then do the comparison.
Also it helps if you could put across a version naming convention across to 
devs.
If i do get the chance, i will attach a patch
```

Original comment by `ashok...@gmail.com` on 27 Dec 2009 at 2:52

**Comment by GoogleCodeExporter** [comment]
```
@Firefart,
Attached is a patch that will work for comparing versions (containing 
alphabets).
Didn't really check if its efficient. Let me know if this works!
Cheers! Happy holidays
/Ash
```

Original comment by `ashok...@gmail.com` on 27 Dec 2009 at 3:38

Attachments:
- [compare_patch.txt](https://storage.googleapis.com/google-code-attachments/cyanogen-updater/issue-155/comment-6/compare_patch.txt)

**Comment by GoogleCodeExporter** [comment]
```
thx its implemented for the next release (with a few changes ;) )
```

Original comment by `FireFart` on 27 Dec 2009 at 8:12
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
Issue 156 has been merged into this issue.
```

Original comment by `FireFart` on 27 Dec 2009 at 10:58

**Comment by GoogleCodeExporter** [comment]
```
Issue 157 has been merged into this issue.
```

Original comment by `FireFart` on 28 Dec 2009 at 8:21

**Comment by GoogleCodeExporter** [comment]
```
Issue 159 has been merged into this issue.
```

Original comment by `FireFart` on 29 Dec 2009 at 1:34

**Comment by GoogleCodeExporter** [comment]
```
Issue 161 has been merged into this issue.
```

Original comment by `FireFart` on 31 Dec 2009 at 10:52

**Comment by GoogleCodeExporter** [comment]
```
Issue 162 has been merged into this issue.
```

Original comment by `FireFart` on 2 Jan 2010 at 8:25

**Comment by GoogleCodeExporter** [comment]
```
Issue 163 has been merged into this issue.
```

Original comment by `FireFart` on 3 Jan 2010 at 12:53

**Comment by GoogleCodeExporter** [comment]
```
Can I suggest that you don't mark this fixed until the fix is released?  Then 
you
would not receive all these duplicates (at least, you would not have received 
mine
because I checked the open issues before submitting... ;-)
```

Original comment by `adam.spi...@gmail.com` on 3 Jan 2010 at 1:03

**Comment by GoogleCodeExporter** [comment]
```
Issue 164 has been merged into this issue.
```

Original comment by `FireFart` on 4 Jan 2010 at 3:11

**Comment by GoogleCodeExporter** [comment]
```
Issue 165 has been merged into this issue.
```

Original comment by `FireFart` on 4 Jan 2010 at 4:45

**Comment by GoogleCodeExporter** [comment]
```
Released Version 4.5.2 to fix this issue
```

Original comment by `FireFart` on 4 Jan 2010 at 5:55

**Comment by GoogleCodeExporter** [comment]
Original comment by `FireFart` on 17 Feb 2010 at 6:38
- Changed state: **Verified**
